From f8332b72714819c6eebc624f423bc2fecd5f238d Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:55:52 -0400
Subject: [scsi] qla2xxx: print MAC via %pMR

Message-id: <1408391817-6389-5-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87924
O-Subject: [PATCH RHEL 7.1 e-stor 04/69] qla2xxx: print MAC via %pMR.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Andy Shevchenko <andriy.shevchenko@linux.intel.com>

scsi commit id ac3d5f0ab9673fc3dc8d73cce38a00222b69dc68

Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_attr.c b/drivers/scsi/qla2xxx/qla_attr.c
index b1d32c4..5f174b8 100644
--- a/drivers/scsi/qla2xxx/qla_attr.c
+++ b/drivers/scsi/qla2xxx/qla_attr.c
@@ -1290,10 +1290,7 @@ qla2x00_vn_port_mac_address_show(struct device *dev,
  if (!IS_CNA_CAPABLE(vha->hw))
   return snprintf(buf, PAGE_SIZE, "\n");
 
- return snprintf(buf, PAGE_SIZE, "%02x:%02x:%02x:%02x:%02x:%02x\n",
-     vha->fcoe_vn_port_mac[5], vha->fcoe_vn_port_mac[4],
-     vha->fcoe_vn_port_mac[3], vha->fcoe_vn_port_mac[2],
-     vha->fcoe_vn_port_mac[1], vha->fcoe_vn_port_mac[0]);
+ return snprintf(buf, PAGE_SIZE, "%pMR\n", vha->fcoe_vn_port_mac);
 }
 
 static ssize_t
-- 
1.7.1